static/js/ducks/sources.js

```diff
@@ -74,6 +102,13 @@ messageHandler.add((actionType, payload, dispatch, getState) => {
     }
   }
 
+  const { gcnEvent } = getState();
+  if (actionType === FETCH_GCNEVENT_SOURCES) {
+    if (gcnEvent?.dateobs) {
+      dispatch(fetchGcnEventSources(gcnEvent?.dateobs));
+    }
+  }
```

Comment:
Wouldn't this message have a payload like `gcnEventId: ...`? If so, we want to check whether the currently loaded `gcnEvent.id` matches the message payload ID, and only do the fetch then. Something like: 

```suggestion
  if (actionType === FETCH_GCNEVENT_SOURCES) {
    if (gcnEvent && gcnEvent.id === payload.gcnEvent.id) {
      dispatch(fetchGcnEventSources(gcnEvent.dateobs));
    }
  }
```